gradoop-flink/src/main/java/org/gradoop/flink/model/impl/operators/propertytransformation/PropertyTransformation.java

```diff
@@ -0,0 +1,163 @@
+ * (0, "Person", {yob: 1970, yob__1: 1973})<br>
+ * (1, "Person", {yob: 1970, yob__1: 1977})<br>
+ * (2, "Person", {yob: 1980, yob__1: 1984})<br>
+ * (3, "Person", {yob: 1980, yob__1: 1989})<br>
+ * <p>
+ * This example shows that this operation may be used as pre processing for the
+ * {@link org.gradoop.flink.model.impl.operators.grouping.Grouping} operator.
+ */
+public class PropertyTransformation implements UnaryGraphToGraphOperator {
+
+  /**
+   * Label of the element whose property shall be transformed.
+   */
+  private String label;
+  /**
+   * Property key.
+   */
+  private String propertyKey;
+  /**
+   * Transformation function which shall be applied to a property of a graph head.
+   */
+  private PropertyTransformationFunction graphHeadTransformationFunction;
+  /**
+   * Transformation function which shall be applied to a property of a vertex.
+   */
+  private PropertyTransformationFunction vertexTransformationFunction;
+  /**
+   * Transformation function which shall be applied to a property of an edge.
+   */
+  private PropertyTransformationFunction edgeTransformationFunction;
+  /**
+   * New property key.
+   */
+  private String newPropertyKey;
+  /**
+   * True, if the history of a property key shall be kept.
+   */
+  private boolean keepHistory;
+
+  /**
+   * Valued constructor.
+   *
+   * @param label                            label of the element whose property shall be
+   *                                         transformed
+   * @param propertyKey                      property key
+   * @param graphHeadTransformationFunction  transformation function which shall be applied to a
+   *                                         property of a graph head
+   * @param vertexTransformationFunction     transformation function which shall be applied to a
+   *                                         property of a vertex
+   * @param edgeTransformationFunction       transformation function which shall be applied to a
+   *                                         property of an edge
+   * @param newPropertyKey                   new property key
+   * @param keepHistory                flag to enable versioning
+   */
+  public PropertyTransformation(String label, String propertyKey,
+      PropertyTransformationFunction graphHeadTransformationFunction,
+      PropertyTransformationFunction vertexTransformationFunction,
+      PropertyTransformationFunction edgeTransformationFunction, String newPropertyKey,
+      boolean keepHistory) {
+    this.label = label;
+    this.propertyKey = propertyKey;
+    this.graphHeadTransformationFunction = graphHeadTransformationFunction;
+    this.vertexTransformationFunction = vertexTransformationFunction;
+    this.edgeTransformationFunction = edgeTransformationFunction;
+    this.newPropertyKey = newPropertyKey;
+    this.keepHistory = keepHistory;
+  }
+
+  @Override
+  public LogicalGraph execute(LogicalGraph graph) {
```

Comment:
Correct me if I'm wrong, but will the following do the same? We can use the element transformation as we discussed. I think we do not need a additional class `PeropertyTransformationBase` with this.
```
public LogicalGraph execute(LogicalGraph graph) {
    return graph.transform(
      graphHeadTransformationFunction == null ? null : new BasePropertyTransformationFunction<>(label, propertyKey, graphHeadTransformationFunction, newPropertyKey, keepHistory),
      vertexTransformationFunction == null ? null : new BasePropertyTransformationFunction<>(label, propertyKey, vertexTransformationFunction, newPropertyKey, keepHistory),
      edgeTransformationFunction == null ? null : new BasePropertyTransformationFunction<>(label, propertyKey, edgeTransformationFunction, newPropertyKey, keepHistory));
  }
```